gitops/argocd/image/register.sh

```diff
@@ -0,0 +1,134 @@
+    # to clarify: is it a directory plnsvc for all the workload clusters or is plnsvc already specific to 1 workload cluster?
+    # If it is the later we need to differentiate between workload clusters (a subdirectory?), kcp and Argo CD credentials
+    # something like:
+    #   credentials/kubeconfig/workload/plnsvc1
+    #   credentials/kubeconfig/workload/plnsvc2
+    #   credentials/kubeconfig/workload/plnsvc3
+    # If it the former, why plnsvc as name? How does it identify the role they are playing within the service?
+    # Looking at /environment it seems to be the former.
+    files=($(ls $DATA_DIR/gitops/credentials/kubeconfig/plnsvc))
+    for kubeconfig in "${files[@]}"; do
+        clusters_sub=($(KUBECONFIG=${DATA_DIR}/gitops/credentials/kubeconfig/plnsvc/${kubeconfig} kubectl config get-clusters | cut -d ':' -f 1 ))
+        clusters_sub=( ${clusters_sub[@]:1} )
+        clusters+=( ${clusters_sub[@]} )
+        for i in "${!clusters_sub[@]}"; do
+            kubeconfigs+=( ${kubeconfig} )
+        done
+    done
+}
+
+parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+
+prechecks
+
+printf "Retrieving clusters\n"
+get_clusters
+
+printf "Logging into Argo CD\n"
+# TODO: there may be a better way than user/password
+# there should be no assumption that Argo CD runs on the same cluster
+argocd ${insecure} login $ARGO_URL --username $ARGO_USER --password $ARGO_PWD
+
+printf "Getting the list of registered clusters\n"
+existing_clusters=$(argocd cluster list -o json | jq '.[].name')
+
+printf "Registering clusters and provisioning credentials for Argo CD\n"
+for i in "${!clusters[@]}"; do
+    printf "Processing cluster %s\n" "${clusters[$i]}"
+    if echo "${existing_clusters}" | grep "${clusters[$i]}"; then
+        printf "Cluster already registered\n"
+    else
+        printf "Registering cluster\n"
+	# TODO: Romain has /environment/plnsvc/cluster-name/config
+	# to clarify:
+	# - do we need "config". It seems I don't
+	# - as above is "plnsvc" the right name? workload or compute may be a better choice.
+	# - the directory name needs to match with the cluster name in kubeconfig
+	# - need to document that a directory /environment/plnsvc/${clusters[$i]}/ns-rbac with a kustomize having
+	# /environment/plnsvc/base/ns-rbac as a base needs to be created for every new cluster
+        # Split between namespace creation and application of rbac policies:
+	# - `argocd cluster add` requires the namespaces to exist
+	# - `argocd cluster add` applies default rbac that may differ from what is desired
+        KUBECONFIG=${DATA_DIR}/gitops/credentials/kubeconfig/plnsvc/${kubeconfigs[$i]} kubectl apply -k ${DATA_DIR}/environment/plnsvc/${clusters[$i]}/namespaces
+        KUBECONFIG=${DATA_DIR}/gitops/credentials/kubeconfig/plnsvc/${kubeconfigs[$i]} argocd ${insecure} cluster add "${clusters[$i]}" --system-namespace argocd-management --namespace=tekton-pipelines --namespace=kcp-syncer
```

Comment:
Currently, clustername is passed for 'argocd add' which is incorrect. We need to add context name. Modified this line to extract context name. But I think you can do this in the above for loop too and store an array of context names too.

```
KUBECONFIG=${DATA_DIR}/gitops/credentials/kubeconfig/plnsvc/${kubeconfigs[$i]} argocd ${insecure} cluster add "$(yq ".contexts[$i].name" <"${DATA_DIR}/gitops/credentials/kubeconfig/plnsvc/${kubeconfigs[$i]}")" --system-namespace argocd-management --namespace=tekton-pipelines --namespace=kcp-syncer --yes
```

Reply:
I modified the script so that "context" is passed to the argocd command. The cluster name is still used as a key for the directory as the context may not be unique across multiple kubeconfigs